cli/crates/cli/tests/utils/environment.rs

```diff
@@ -49,42 +52,121 @@ fn get_free_port() -> u16 {
-            endpoint: format!("http://127.0.0.1:{port}/graphql"),
-            playground_endpoint: format!("http://127.0.0.1:{port}"),
-            schema_path,
+            endpoint,
+            playground_endpoint,
+            directory,
+            port,
             temp_dir,
+            schema_path,
+            commands,
+            dynamodb_env,
+        }
+    }
+
+    #[cfg(not(feature = "dynamodb"))]
+    fn init_internal(port: u16) -> Self {
+        let temp_dir = Arc::new(tempdir().unwrap());
+        env::set_current_dir(temp_dir.path()).unwrap();
+        let schema_path = temp_dir
+            .path()
+            .join(GRAFBASE_DIRECTORY_NAME)
+            .join(GRAFBASE_SCHEMA_FILE_NAME);
+        let directory = temp_dir.path().to_owned();
+        println!("Using temporary directory {:?}", directory.as_os_str());
+        let commands = vec![];
+        let endpoint = format!("http://127.0.0.1:{port}/graphql");
+        let playground_endpoint = format!("http://127.0.0.1:{port}");
+        Self {
+            endpoint,
+            playground_endpoint,
+            directory,
             port,
+            temp_dir,
+            schema_path,
+            commands,
         }
     }
 
+    /// Same environment but different port.
+    #[allow(clippy::needless_return)]
     pub fn from(other: &Environment) -> Self {
         let port = get_free_port();
-
         let temp_dir = other.temp_dir.clone();
-
-        Self {
-            directory: other.directory.clone(),
-            commands: vec![],
-            endpoint: format!("http://127.0.0.1:{port}/graphql"),
-            playground_endpoint: format!("http://127.0.0.1:{port}"),
-            schema_path: other.schema_path.clone(),
-            temp_dir,
-            port,
-        }
+        let endpoint = format!("http://127.0.0.1:{port}/graphql");
+        let playground_endpoint = format!("http://127.0.0.1:{port}");
+        let commands = vec![];
+        cfg_if!(
+            if #[cfg(feature = "dynamodb")] {
+                return Self {
+                    directory: other.directory.clone(),
+                    commands,
+                    endpoint,
+                    playground_endpoint,
+                    schema_path: other.schema_path.clone(),
+                    temp_dir,
+                    port,
+                    dynamodb_env: dynamodb::DynamoDbEnvironment {
+                        dynamodb_client: None, // Only one dynamodb client is needed for the cleanup.
+                        table_name: other.dynamodb_env.table_name.clone(),
+                    },
+                };
+            } else {
+                return Self {
+                    directory: other.directory.clone(),
+                    commands,
+                    endpoint,
+                    playground_endpoint,
+                    schema_path: other.schema_path.clone(),
+                    temp_dir,
+                    port,
+                };
+            }
```

Comment:
Could this just be a `#[cfg` on the `dynamodb_env` field rather than duplicating everything?

```suggestion
        Self {
            directory: other.directory.clone(),
            commands,
            endpoint,
            playground_endpoint,
            schema_path: other.schema_path.clone(),
            temp_dir,
            port,
            #[cfg(feature = "dynamodb")]
            dynamodb_env: dynamodb::DynamoDbEnvironment {
                dynamodb_client: None, // Only one dynamodb client is needed for the cleanup.
                table_name: other.dynamodb_env.table_name.clone(),
            }
        }
```

Reply:
Nice, applied!